src/parser/Statement.ts

```diff
@@ -591,6 +591,37 @@ export class PrintStatement extends Statement {
     }
 }
 
+export class DimStatement extends Statement {
+    constructor(
+        public dimToken: Token,
+        public identifier?: Identifier,
+        public openingSquare?: Token,
+        public dimensions?: Expression[],
+        public closingSquare?: Token
+    ) {
+        super();
+        this.range = util.createRangeFromPositions(
+            this.dimToken.range.start,
+            (this.closingSquare ?? this.dimensions[this.dimensions.length - 1] ?? this.openingSquare ?? this.identifier ?? this.dimToken).range.end
+        );
+    }
+    public range: Range;
+
+    public transpile(state: TranspileState) {
```

Comment:
Please add some tests proving that this transpiles properly. You can see some transpile test examples [here](https://github.com/rokucommunity/brighterscript/blob/45a31b18b10fecf6ed1046e4e249c5df5a4fcc66/src/files/BrsFile.spec.ts#L1551)